examples/keep-alive-framework/src/main/scala/com/mesosphere/usi/helloworld/KeepAliveMesosClientFactory.scala

```diff
@@ -3,32 +3,29 @@ package com.mesosphere.usi.helloworld
 import akka.actor.ActorSystem
 import akka.stream.ActorMaterializer
 import akka.stream.scaladsl.Sink
-import com.mesosphere.mesos.client.MesosClient
+import com.mesosphere.mesos.client.{CredentialsProvider, MesosClient}
 import com.mesosphere.mesos.conf.MesosClientSettings
-import com.typesafe.config.Config
 import org.apache.mesos.v1.Protos.FrameworkInfo
 
 import scala.concurrent.duration._
 import scala.concurrent.Await
-import scala.sys.SystemProperties
 
 /**
-  * Helper that builds a mesos client that can be used by USI
+  * Helper that builds a Mesos client that can be used by USI
   */
-class KeepAliveMesosClientFactory(conf: Config)(implicit system: ActorSystem, mat: ActorMaterializer) {
+class KeepAliveMesosClientFactory(settings: MesosClientSettings, authorization: Option[CredentialsProvider])(
+    implicit system: ActorSystem,
+    mat: ActorMaterializer) {
 
-  val settings = MesosClientSettings.fromConfig(conf)
   val frameworkInfo = FrameworkInfo
     .newBuilder()
-    .setUser(
-      new SystemProperties()
-        .get("user.name")
-        .getOrElse(throw new IllegalArgumentException("A local user is needed to launch Mesos tasks")))
+    .setUser("root")
     .setName("KeepAliveFramework")
-    .addRoles("test")
+    .addRoles("slave_public")
```

Comment:
why do you need to use `slave_public`?

Reply:
That's a left over from my tests. Offer accepts would not get through.